ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/StateManager.java

```diff
@@ -1468,6 +1476,7 @@ final void destroyed ()
     protected Hashtable usingActions = null;
     protected final Uid objectUid;
     protected int objectModel = ObjectModel.SINGLE;
+    protected long birthDate = System.currentTimeMillis();
```

Comment:
@graben since this is now an instance variable with greater visibility what do you think about renaming it to `creationTime` which is more standard than birthDate (which will mean changing the name of the `setter` as well).

Reply:
Good idea, will change it as soon as possible. :-)

Reply:
> Good idea, will change it as soon as possible. :-) Btw getCreationTime should not return String as mentioned in your examples?

Why not? The actual returned string is `new SimpleDateFormat("E, dd MMM yyyy HH:mm:ss Z").format(new Date(birthDate));`

Reply:
The question was before I decided to rename to the variable. I think it's better to return the long value rather than the String representation if further execution is done with the date value!